test/e2e/validation_test.go

```diff
@@ -667,22 +665,24 @@ func (tc *testContext) findNodeCSRs(nodeName string) ([]certificates.Certificate
 
 // validateUpgradeableCondition ensures that the operator's Upgradeable condition is correctly communicated to OLM
 func (tc *testContext) validateUpgradeableCondition(expected meta.ConditionStatus) error {
-	ocName, present := os.LookupEnv(condition.OperatorConditionName)
-	if !present {
-		// Implies operator is not OLM-managed
-		return nil
-	}
-	err := wait.Poll(retry.Interval, retry.ResourceChangeTimeout, func() (bool, error) {
-		oc := &operators.OperatorCondition{}
-		err := tc.client.Cache.Get(context.TODO(), types.NamespacedName{Namespace: tc.namespace, Name: ocName}, oc)
+	deployment, err := tc.client.K8s.AppsV1().Deployments(resourceNamespace).Get(context.TODO(), resourceName,
+		meta.GetOptions{})
+	if err != nil {
+		log.Printf("error getting operator deployment: %v", err)
+		return err
+	}
+	// Get the operator condition name using the deployment spec
+	ocName := deployment.Spec.Template.Spec.Containers[0].Env[3].Value
```

Comment:
Is it safe to get these values by index? Will the operator condition name always be in the third position?

Reply:
that has been my observation but I think it will be safer to loop through and find the env var by key, will update.

Reply:
Also the `Container[0]` -- is it worth looping over containers and filtering for the the desired one by name?